lib/acts_as_ar_model.rb

```diff
@@ -142,8 +142,30 @@ def self.reflections
   # Find routines
   #
 
+  # This method is called by QueryRelation upon executing the query.
+  #   Since it will receive non-legacy search options, we need to convert
+  #   them to handle the legacy find.
+  def self.search(mode, options = {})
```

Comment:
So what I see you proposing is essentially creating an anonymous proc for the callback instead of a method / telling QueryRelation to use that method.

The only thing I don't like about the anonymous proc is that it's not testable in isolation.